prow/crier/reporters/gcs/internal/util/util.go

```diff
@@ -47,27 +48,30 @@ func (sa StorageAuthor) NewWriter(ctx context.Context, bucket, path string, over
 	if err != nil {
 		return nil, err
 	}
-	return sa.Opener.Writer(ctx, fmt.Sprintf("%s://%s/%s", pp.StorageProvider(), pp.Bucket(), path), pkgio.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(true)})
+	return sa.Opener.Writer(ctx, fmt.Sprintf("%s://%s/%s", pp.StorageProvider(), pp.Bucket(), path),
+		pkgio.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(!overwrite)})
```

Comment:
This bug, apparently introduced a month ago, will definitely have broken things! In particular IIRC we depend on this for writing prowjob.json to behave properly. The inverse breakage would've been much worse, though. 😅 

It might be nice to see some sort of test for this (but I don't think that should block anything here — just an observation).

Reply:
I checked the code and feel it not quite straightforward to add unit tests for this function, as it uses `io.Opener` which we don't have a fake implementation.. 

So I would prefer to keep it as is, since it should not be that easy to break in the future as this line of code is not as long as before.